scripts/deployment.ts

```diff
@@ -0,0 +1,156 @@
+import { ethers } from "hardhat";
+import { ContractFactory, Contract, providers, Wallet } from "ethers";
+import "@nomiclabs/hardhat-ethers";
+
+
+async function deployContract(contractName: string, signer: any): Promise<Contract> {
+  const factory: ContractFactory = await ethers.getContractFactory(contractName);
+  const connectedFactory: ContractFactory = factory.connect(signer);
+  const contract: Contract = await connectedFactory.deploy();
+  await contract.deployTransaction.wait();
+  console.log(`New ${contractName} created at address:`, contract.address);
+  return contract;
+}
+
+async function deployAllContracts(signer: any): Promise<Record<string, Contract>> {
+  const bridgeCreator = await deployContract("BridgeCreator", signer);
+  const prover0 = await deployContract("OneStepProver0", signer);
+  const proverMem = await deployContract("OneStepProverMemory", signer);
+  const proverMath = await deployContract("OneStepProverMath", signer);
+  const proverHostIo = await deployContract("OneStepProverHostIo", signer);
+  const OneStepProofEntryFactory: ContractFactory = await ethers.getContractFactory("OneStepProofEntry");
+  const OneStepProofEntryFactoryWithDeployer: ContractFactory = OneStepProofEntryFactory.connect(signer);
+  const osp: Contract = await OneStepProofEntryFactoryWithDeployer.deploy(
+    prover0.address,
+    proverMem.address,
+    proverMath.address,
+    proverHostIo.address
+  );
+  await osp.deployTransaction.wait();
+  console.log("New osp created at address:", osp.address);
+  const challengeManager = await deployContract("ChallengeManager", signer);
+  const rollupAdmin = await deployContract("RollupAdminLogic", signer);
+  const rollupUser = await deployContract("RollupUserLogic", signer);
+  const validatorUtils = await deployContract("ValidatorUtils", signer);
+  const validatorWalletCreator = await deployContract("ValidatorWalletCreator", signer);
+  const rollupCreator = await deployContract("RollupCreator", signer);
+  return {
+    bridgeCreator,
+    prover0,
+    proverMem,
+    proverMath,
+    proverHostIo,
+    osp,
+    challengeManager,
+    rollupAdmin,
+    rollupUser,
+    validatorUtils,
+    validatorWalletCreator,
+    rollupCreator,
+  };
+}
+
+async function main() {
+
+  // Get the private key and RPC to deploy the contract
+  const privateKey = process.env.PRIVATE_KEY;
+  if (!privateKey) {
+    throw new Error("Private key is not defined");
+  }
+
+  const providerAPI = process.env.RPC_URL;
+  if (!providerAPI) {
+    throw new Error("RPC URL is not defined");
+  }
+
+  const provider = new providers.JsonRpcProvider(providerAPI)
+  const signer = new Wallet(privateKey, provider)
```

Comment:
it appear that this script is expected to be called via hardhat, then why don't we use the hardhat network and signer?

Reply:
I think we want to use it to deploy on sepolia. Am I wrong?

Reply:
It doesn't matter, you can still config network in hardhat